src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1387,17 +1403,27 @@ private function dumpValue($value, $interpolate = true)
 
             return sprintf('array(%s)', implode(', ', $code));
         } elseif ($value instanceof IteratorArgument) {
+            $countCode = array();
+            $countCode[] = 'function() {';
+            $operands = array(0);
+
             $code = array();
             $code[] = 'new RewindableGenerator(function() {';
-            foreach ($value->getValues() as $k => $v) {
-                $v = $this->wrapServiceConditionals($v, sprintf("        yield %s => %s;\n", $this->dumpValue($k, $interpolate), $this->dumpValue($v, $interpolate)));
+            foreach ($value->getValues() as $k => $value) {
+                $v = $this->wrapServiceConditionals($value, sprintf("        yield %s => %s;\n", $this->dumpValue($k, $interpolate), $this->dumpValue($value, $interpolate)));
                 foreach (explode("\n", $v) as $v) {
                     if ($v) {
                         $code[] = '    '.$v;
                     }
                 }
+
+                ($c = $this->getServiceConditionals($value)) ? $operands[] = "(int) ($c)" : ++$operands[0];
```

Comment:
If there are several conditions, we'll get something like:

```php
(int) ($this->has('foo') && $this->has('other'))
```

for a single argument, so should be ok, right? (Actually I don't even know why you would have several conditions for a single service. In case of a collection of service all being ignored on invalid reference?)